src/views/Collectibles/hooks/useBunnySpecialLottery.tsx

```diff
@@ -0,0 +1,217 @@
+const NO_CLAIM: NftClaim = {
+  canClaim: false,
+  mintData: null,
+}
+
+const useBunnySpecialLottery = (): {
+  canClaimLottie: () => Promise<NftClaim>
+  canClaimLucky: () => Promise<NftClaim>
+  canClaimBaller: () => Promise<NftClaim>
+} => {
+  const { account } = useWeb3React()
+  const lotteryNftContract = useBunnySpecialLotteryContract()
+  const lotteryNftContractAddress = getBunnySpecialLotteryAddress()
+  const currentLotteryId = useGetCurrentLotteryId()
+  const currentLotteryIdNum = parseInt(currentLotteryId)
+  const [publicContractData, setPublicContractData] = useState<PublicContractData>(null)
+  const [userRounds, setUserRounds] = useState<UserRound[]>(null)
+
+  useEffect(() => {
+    setUserRounds(null)
+  }, [account])
+
+  useEffect(() => {
+    const getUserData = async () => {
+      const { startLotteryRound, finalLotteryRound } = publicContractData
+      const startLotteryRoundInt = parseInt(startLotteryRound)
+      const finalLotteryRoundInt = parseInt(finalLotteryRound)
+
+      const lotteryIdsArray = []
+      for (let i = startLotteryRoundInt; i <= finalLotteryRoundInt; i++) {
+        lotteryIdsArray.push(`${i}`)
+      }
+      const userParticipation = await getGraphLotteryUser(account, undefined, undefined, {
+        lottery_in: lotteryIdsArray,
+      })
+      setUserRounds(userParticipation.rounds)
+    }
+
+    const getPublicContractData = async () => {
+      const calls = [
+        {
+          name: 'startLotteryRound',
+          address: lotteryNftContractAddress,
+          params: [],
+        },
+        {
+          name: 'finalLotteryRound',
+          address: lotteryNftContractAddress,
+          params: [],
+        },
+      ]
+      try {
+        const [[startRound], [finalRound]] = await multicallv2(bunnySpecialLotteryAbi, calls)
+        const startRoundAsString = ethersToSerializedBigNumber(startRound)
+        const finalRoundAsString = ethersToSerializedBigNumber(finalRound)
+        setPublicContractData({ startLotteryRound: startRoundAsString, finalLotteryRound: finalRoundAsString })
+      } catch (error) {
+        console.error('Failed to fetch public lottery nft data', error)
+      }
+    }
+
+    if (account) {
+      if (!publicContractData) {
+        getPublicContractData()
+      }
+      if (currentLotteryIdNum && publicContractData && !userRounds) {
+        getUserData()
+      }
+    }
+  }, [currentLotteryIdNum, userRounds, publicContractData, lotteryNftContractAddress, account])
+
+  // If user has not entered the right rounds, or has already claimed NFT - generic checks false
+  const genericClaimChecks = useCallback(
+    async (variationId: number | string): Promise<boolean> => {
+      let hasClaimed = false
+      if (!userRounds) {
+        return false
+      }
+      try {
+        hasClaimed = await lotteryNftContract.hasClaimed(account, variationId)
+      } catch (error) {
+        console.error(`Failed to check hasClaimed for id ${variationId}`, error)
```

Comment:
What happens if this error is thrown? Maybe return true or false?

Reply:
Nice catch 👍